### PR TITLE
Custom query fix

### PR DESCRIPTION
**Pull request by mhavey**
Pull request for RDF connector changes

### CODE REVIEW

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java
Comment (abhishekpradeepmishra) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java:1]: remove this

//Client client = null; NOT USED

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class NeptuneConnection
 {
     private static Cluster cluster = null;
 
-    NeptuneConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM)
+    protected NeptuneConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM, String region)
```
Comment (abhishekpradeepmishra) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java]: where is the region parameter used?

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    NeptuneSparqlRepository neptuneSparqlRepo = null;
+    RepositoryConnection connection = null;
+    TupleQueryResult queryResult = null;
+
+    public NeptuneSparqlConnection(String neptuneEndpoint, String neptunePort, boolean enabledIAM, String region) 
+    {
+        super(neptuneEndpoint, neptunePort, enabledIAM, region);
+        this.connectionString = "https://" + neptuneEndpoint + ":" + neptunePort;
+        this.enabledIAM = enabledIAM;
+        this.region = region;
```
Comment (abhishekpradeepmishra) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java]: won't this be available from base class

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        if (this.enabledIAM) {
+            logger.info("Connecting with IAM auth to " + this.connectionString);
+            final AWSCredentialsProvider awsCredentialsProvider = new DefaultAWSCredentialsProviderChain();
+            try {
+                this.neptuneSparqlRepo = new NeptuneSparqlRepository(this.connectionString, awsCredentialsProvider, this.region);
+            } 
+            catch (NeptuneSigV4SignerException e) {
+                e.printStackTrace();
+                logger.error("Error connecting with IAM auth", e);
+            }
```
Comment (abhishekpradeepmishra) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java]: Don't catch the exception here. Move all exception catch code to NeptuneRecordHandler

File: athena-neptune/docs/aws-glue-sample-scripts/RDF.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+- **Query-based**: The table represents the resultset of a SPARQL query. Each row is one result. See the route_rdf example below.
+
+In each case, you define columns and use table properties to map RDF to that column structure. Here is a summary of table properties to indicate RDF mapping:
+
+|Property|Values|Description|
+|--------|------|-----------|
+|componenttype|rdf||
+|querymode|class, sparql||
+|sparql|SPARQL query to use to find resultset.|Only if querymode='sparql'. Omit prefixes. Define prefixes as table properties.|
+|classuri|Class of resources to find|In curie form prefix:classname. Only if querymode='class'. Connector will query for resources whose RDF type is this classuri.|
```
Comment (abhishekpradeepmishra) [inline athena-neptune/docs/aws-glue-sample-scripts/RDF.md:15]: is this intended spelling?

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                }
+            }
+            sparql += " }";
+        } 
+        else {
+            throw new RuntimeException("Illegal RDF params");
+        }
+
+        // 3. Add filters. This means adding FILTER conditions to the SPARQL.
+        // No longer used
```
Comment (abhishekpradeepmishra) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java]: remove this

**Comment by abhishekpradeepmishra** [comment]
I have added comments, kindly review them.

I will test the connector next week

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh [hunk 6]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# id is the reserved column for vertex tables
+# id, from, to are the reserved columns for edge tables
+
+echo $1;
+echo $2;
+echo $3;
```
Comment (abhishekpradeepmishra) [inline athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: remove the 3rd parameter, its not needed and update the script accordingly

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            "prefix_class":"http://kelvinlawrence.net/air-routes/class/",
+            "querymode":"class",
+            "classuri":"class:Airport",
+            "subject":"id",
+            "strip_uri":"true",
+            "preds_prefix":"prop"
+        } 
+        }' \
+    --profile $1 \
+    --endpoint https://glue.$2.amazonaws.com
```
Comment (abhishekpradeepmishra) [inline athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: missing back slash, script breaks

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            "componenttype":"rdf",
+            "prefix_prop":"http://kelvinlawrence.net/air-routes/datatypeProperty/",
+            "prefix_op":"http://kelvinlawrence.net/air-routes/objectProperty/",
+            "querymode":"sparql",
+            "sparql": "select ?incode ?outcode ?dist where {  ?resin op:route ?resout . GRAPH ?route { ?resin op:route ?resout } .  ?route prop:dist ?dist  . ?resin prop:code ?incode .?resout prop:code ?outcode . } ",
+            "strip_uri":"true",
+        } 
+        }' \
+    --profile $1 \
+    --endpoint https://glue.$2.amazonaws.com
```
Comment (abhishekpradeepmishra) [inline athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: missing back slash, script breaks

Comment (abhishekpradeepmishra) [reply athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: '--table-input': Invalid JSON: Expecting property name enclosed in double quotes: line 16 column 9 (char 813)
JSON received: {"Name":"airport_route", "StorageDescriptor":{ 
        "Columns":[

File: athena-neptune/docs/aws-glue-sample-scripts/RDF.md
Comment (abhishekpradeepmishra) [inline athena-neptune/docs/aws-glue-sample-scripts/RDF.md:1]: run spell check on the file

File: athena-neptune/docs/aws-glue-sample-scripts/RDF.md (hunk 4 above)
Comment (abhishekpradeepmishra) [reply athena-neptune/docs/aws-glue-sample-scripts/RDF.md:15]: mark config options like preds_prefix as optional

File: athena-neptune/athena-neptune.yaml [hunk 9]
```diff
@@ -25,7 +25,7 @@ Parameters:
     Description: 'To find the Neptune cluster resource ID in the Amazon Neptune AWS Management Console, choose the DB cluster that you want. The Resource ID is shown in the Configuration section.'
     Type: String
   NeptuneGraphType:
-    Description: 'Type of graph created in Neptune, defaults to PROPERTYGRAPH. RDF support is yet to be implemented'
+    Description: 'Type of graph created in Neptune, defaults to PROPERTYGRAPH. Allowed values: PROPERTYGRAPH, RDF'
```
Comment (abhishekpradeepmishra) [inline athena-neptune/athena-neptune.yaml:28]: add this as allowed list values

"AllowedValues" : ["PROPERTYGRAPH", "RDF"],

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java
Comment (mhavey) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java:1]: agreed

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java (hunk 1 above)
Comment (mhavey) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java]: sparql connection needs region

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java (hunk 5 above)
Comment (mhavey) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/RDFHandler.java]: agreed

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java (hunk 2 above)
Comment (mhavey) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java]: modified

File: athena-neptune/athena-neptune.yaml (hunk 9 above)
Comment (mhavey) [reply athena-neptune/athena-neptune.yaml:28]: added

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh (hunk 6 above)
Comment (mhavey) [reply athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: removed

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh (hunk 8 above)
Comment (mhavey) [reply athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: I'll test it

File: athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh (hunk 7 above)
Comment (mhavey) [reply athena-neptune/docs/aws-glue-sample-scripts/manual/sample-cli-rdf.sh]: done

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java (hunk 3 above)
Comment (mhavey) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/rdf/NeptuneSparqlConnection.java]: I will keep some catch blocks -- for the safeClose() methods. But this one I will bubble up

**Pull request closed without merging**